test/cli/testutils.py

```diff
@@ -72,8 +75,82 @@ def __lookup_cppcheck_exe():
     return exe_path
 
 
+def __run_subprocess_tty(args, env=None, cwd=None, timeout=None):
```

Comment:
Still need to review this. Everything else LGTM.

Reply:
I have not verified all the behavior of the calls in detail but it looks fine. Also it is only used by the added tests and the CI works so even if there is a small oversight it would be very limited.